opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/_load.py

```diff
@@ -0,0 +1,123 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from logging import getLogger
+from os import environ
+
+from pkg_resources import iter_entry_points
+
+from opentelemetry.instrumentation.dependencies import (
+    get_dist_dependency_conflicts,
+)
+from opentelemetry.instrumentation.distro import BaseDistro, DefaultDistro
+from opentelemetry.instrumentation.environment_variables import (
+    OTEL_PYTHON_CONFIGURATOR,
+    OTEL_PYTHON_DISABLED_INSTRUMENTATIONS,
+    OTEL_PYTHON_DISTRO,
+)
+from opentelemetry.instrumentation.version import __version__
+
+logger = getLogger(__name__)
+
+
+def _load_distros() -> BaseDistro:
+    distro_name = environ.get(OTEL_PYTHON_DISTRO, None)
+    for entry_point in iter_entry_points("opentelemetry_distro"):
+        try:
+            if distro_name is None or distro_name == entry_point.name:
```

Comment:
Should we call `entry_point.load()` if `distro_name is None`?

Reply:
This seems to make the default behavior to be "return the first distro if there is no selected distro. This is a behavior we want to avoid, if the distro is not specifically selected a default fixed distro should be returned. Maybe this is what happens because a default distro is always set as the first distro but this function does not make that very obvious.

Reply:
This is the current behavior. I figured it would be best to expand upon it so as to not break existing users. I can add a comment to make that clear in the code, but I feel like it is clearer in the README.